src/openms/source/MATH/STATISTICS/PosteriorErrorProbabilityModel.cpp

```diff
@@ -574,5 +580,262 @@ namespace OpenMS
 
     }
 
+    double PosteriorErrorProbabilityModel::transformScore_(const String & engine, const PeptideHit & hit)
+    {
+      // Set fixed e-value threshold
+      const double smallest_e_value_ = numeric_limits<double>::denorm_min();
+
+      if (engine == "OMSSA") { return (-1) * log10(max(hit.getScore(), smallest_e_value_)); }
+      else if (engine == "MYRIMATCH" ) { return hit.getScore(); }
+      else if (engine == "XTANDEM") { return (-1) * log10(max((double)hit.getMetaValue("E-Value"), smallest_e_value_)); }
+      else if (engine == "MASCOT")
+      {
+        // issue #740: unable to fit data with score 0
+        if (hit.getScore() == 0.0) 
+        {
+          return numeric_limits<double>::quiet_NaN();
+        }
+        // end issue #740
+        if (hit.metaValueExists("EValue"))
+        {
+          return (-1) * log10(max((double)hit.getMetaValue("EValue"), smallest_e_value_));
+        }
+        if (hit.metaValueExists("expect"))
+        {
+          return (-1) * log10(max((double)hit.getMetaValue("expect"), smallest_e_value_));
+        }
+      }
+      else if (engine == "SPECTRAST") { return 100 * hit.getScore(); } // f-val
+      else if (engine == "SIMTANDEM")
+      {
+        if (hit.metaValueExists("E-Value"))
+        {
+          return (-1) * log10(max((double)hit.getMetaValue("E-Value"), smallest_e_value_));
+        }
+      }
+      else if ((engine == "MSGFPLUS") || (engine == "MS-GF+"))
+      {
+        if (hit.metaValueExists("MS:1002053"))  // name: MS-GF:EValue
+        {
+          return (-1) * log10(max((double)hit.getMetaValue("MS:1002053"), smallest_e_value_));
+        }
+        else if (hit.metaValueExists("expect"))
+        {
+          return (-1) * log10(max((double)hit.getMetaValue("expect"), smallest_e_value_));
+        }
+      }
+      else if (engine == "COMET")
+      {
+        if (hit.metaValueExists("MS:1002257")) // name: Comet:expectation value
+        {
+          return (-1) * log10(max((double)hit.getMetaValue("MS:1002257"), smallest_e_value_));
+        }
+        else if (hit.metaValueExists("expect"))
+        {
+          return (-1) * log10(max((double)hit.getMetaValue("expect"), smallest_e_value_));
+        }
+      }
+      else
+      {
+        throw Exception::UnableToFit(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No parameters for chosen search engine", "The chosen search engine is currently not supported");
+      }
+
+    // avoid compiler warning (every code path must return a value, even if there is a throw() somewhere)
+      return std::numeric_limits<double>::max();
```

Comment:
you can probably remove this, if the above throw is taken out of the `else { }`?